README.md

```diff
@@ -589,6 +590,43 @@ Parameters:
         -   `ret` is the result
         -   If no callback is provided, error will be shown as a VSCode notification.
 
+### vscode.with_insert(callback)
+
+Perform operations in insert mode. If in visual mode, this function will **preserve the selection** after switching to
+insert mode.
+
+Parameters:
+
+-   `callback` (function): Callback function to run after switching to insert mode
+
+Examples:
+
+-   Make `editor.action.addSelectionToNextFindMatch` work correctly in any mode.
+
+    ```lua
+    vim.keymap.set({ "n", "x", "i" }, "<C-d>", function()
+      vscode.with_insert(function()
+        vscode.action("editor.action.addSelectionToNextFindMatch")
+      end)
+    end)
+    ```
+
+    ![select-next](https://github.com/vscode-neovim/vscode-neovim/assets/47070852/5a93c87e-626a-4a70-a9ef-5084f747c7ef)
+
+-   Make <kbd>Tab</kbd> work correctly in VSCode snippet mode.
```

Comment:
How does this make Tab work correctly?

Reply:
I understand now, but it should be worded differently. It should be something like "support snippet navigation after a refactor".

Reply:
All right. This is indeed an indirect benefit. Because it is in insert mode, it is normal.